### PR TITLE
Fixes #1594 - Help context is incorrect for deploy dialog

### LINKED ISSUE #1594: Help context is incorrect for deploy dialog
The Eclipse help context is set to `com.google.cloud.tools.eclipse.appengine.deploy.ui.DeployAppEngineStandardProjectContext` in the parent class. It should be set to either a general deploy help ID or the child pages should have their own respective help IDs.

### PR DESCRIPTION
**Pull request by akerekes**

### CODE REVIEW

File: plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/helpContexts.xml [hunk 1]
```diff
@@ -2,15 +2,15 @@
 <contexts>
   <context id="DeployAppEngineStandardProjectContext" title="App Engine Deployment - Standard Environment">
     <!-- Eclipse requires escaped pseudo-HTML here -->
-    <description>Properties used when deploying the project to the App Engine standard environment.
+    <description>Properties used when deploying the project to the App Engine Standard environment.
```
Comment (elharo) [inline plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/helpContexts.xml]: I believe "standard" is customarily lower case in non-title text. See https://cloud.google.com/appengine/docs/standard/ for example

File: plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/helpContexts.xml [hunk 2]
```diff
@@ -2,15 +2,15 @@
...
      
      &lt;b>Account&lt;/b>
      Google user account with permissions to deploy the project.
      
-     &lt;b>Project ID&lt;/b>
-     The customized name you chose when you created the Cloud Platform project in the Google Cloud Console.
+     &lt;b>Project&lt;/b>
+     The Google Cloud Platform project that will be the target for the deployment. If you want to create a new project, you can click on the link to open the Cloud Console.
```
Comment (elharo) [inline plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/helpContexts.xml]: Present tense, "The Google Cloud Platform project that is the deployment target" 
"you can click on the link to open the Cloud Console" --> "follow the link open the Cloud Console"

File: plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/helpContexts.xml [hunk 3]
```diff
@@ -23,4 +23,16 @@
      The Google Cloud Storage bucket used to stage files associated with the deployment. If this argument is not specified, the application's default code bucket is used.
     </description>
   </context>
+
+  <context id="DeployAppEngineFlexProjectContext" title="App Engine Deployment - Flexible Environment">
+    <!-- Eclipse requires escaped pseudo-HTML here -->
+    <description>Properties used when deploying the project to the App Engine Flexible environment.
```
Comment (elharo) [inline plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/helpContexts.xml]: Flexible --> flexible

File: plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/helpContexts.xml [hunk 4]
```diff
@@ -23,4 +23,16 @@
...
+
+     &lt;b>App Engine configuration folder&lt;/b>
```
Comment (elharo) [inline plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/helpContexts.xml]: delete "the various"

File: plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/helpContexts.xml [hunk 5]
```diff
@@ -23,4 +23,16 @@
...
+     The folder that contains the various configuration files used for the deployment, e.g. the app.yml.
```
Comment (elharo) [inline plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/helpContexts.xml]: e.g. the app.yml --> such as app.yaml

(rule is to avoid Latin)

File: plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/helpContexts.xml [hunk 6]
```diff
@@ -23,4 +23,16 @@
...
+
+     &lt;b>Docker file location&lt;/b>
+     In case of a custom runtime, the path of the Docker image file to be used.
```
Comment (elharo) [inline plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/helpContexts.xml]: perhaps "the path to the Docker image to be used for a custom runtime"

File: plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/DeployPreferencesPanel.java [hunk 7]
```diff
@@ -65,4 +65,5 @@ protected void onDispose() {
     }
   }
 
+  abstract protected String getHelpContextId();
```
Comment (elharo) [inline plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/DeployPreferencesPanel.java]: protected abstract

File: plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui.test/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/BlankDeployPreferencesPanelTest.java [hunk 8]
```diff
@@ -0,0 +1,17 @@
+package com.google.cloud.tools.eclipse.appengine.deploy.ui;
```
Comment (elharo) [inline plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui.test/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/BlankDeployPreferencesPanelTest.java]: copyright

File: plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui.test/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/FlexDeployPreferencesPanelTest.java [hunk 9]
```diff
@@ -0,0 +1,32 @@
+package com.google.cloud.tools.eclipse.appengine.deploy.ui;
```
Comment (elharo) [inline plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui.test/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/FlexDeployPreferencesPanelTest.java]: copyright

File: plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/helpContexts.xml [hunk 10]
```diff
@@ -23,4 +23,16 @@
      The Google Cloud Storage bucket used to stage files associated with the deployment. If this argument is not specified, the application's default code bucket is used.
     </description>
   </context>
+
+  <context id="DeployAppEngineFlexProjectContext" title="App Engine Deployment - Flexible Environment">
+    <!-- Eclipse requires escaped pseudo-HTML here -->
+    <description>Properties used when deploying the project to the App Engine flexible environment.
+
+     &lt;b>App Engine configuration folder&lt;/b>
```
Comment (elharo) [inline plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/helpContexts.xml]: Do we call it a folder or a directory?

File: plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui.test/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/FlexDeployPreferencesPanelTest.java (hunk 9 above)
Comment (akerekes) [inline plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui.test/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/FlexDeployPreferencesPanelTest.java]: done

File: plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui.test/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/BlankDeployPreferencesPanelTest.java (hunk 8 above)
Comment (akerekes) [inline plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui.test/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/BlankDeployPreferencesPanelTest.java]: done

File: plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/helpContexts.xml (hunk 10 above)
Comment (akerekes) [inline plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/helpContexts.xml]: The UI uses folder. But based on this: http://stackoverflow.com/questions/5078676/what-is-the-difference-between-a-directory-and-a-folder we should use directory.

File: plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui.test/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/BlankDeployPreferencesPanelTest.java (hunk 8 above)
Comment (elharo) [inline plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui.test/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/BlankDeployPreferencesPanelTest.java]: do you need to push this one?

File: plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui.test/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/FlexDeployPreferencesPanelTest.java (hunk 9 above)
Comment (elharo) [inline plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui.test/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/FlexDeployPreferencesPanelTest.java]: push?

Comment (akerekes) [inline plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui.test/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/FlexDeployPreferencesPanelTest.java]: sorry, I've added licenses to 4 other files, but not to these. I've just added to these as well.

File: plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/helpContexts.xml (hunk 10 above)
Comment (elharo) [inline plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/helpContexts.xml]: so do we want to change this here?

Comment (akerekes) [inline plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/helpContexts.xml]: I file an issue to change the UI and the help together.

Comment (akerekes) [inline plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/helpContexts.xml]: #1630

File: plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/FlexDeployPreferencesPanel.java [hunk 11]
```diff
@@ -164,9 +171,12 @@ private void updateControls() {
 
   private void applyPreferences(FlexDeployPreferences preferences) {
     useValuesButton.setSelection(preferences.getUseDeploymentPreferences());
-    gaeConfigFolderText.setText(preferences.getAppEngineDirectory());
```
Comment (chanseokoh) [inline plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/FlexDeployPreferencesPanel.java]: I'm trying to understand this. Why is this removed?

Comment (akerekes) [inline plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/FlexDeployPreferencesPanel.java]: I have no idea how this happened, I must have deleted it accidentally. Thanks for catching it!
